packages/client/test/integration/mocks/network.ts

```diff
@@ -45,13 +45,16 @@ export function destroyStream(id: string, location: string) {
   }
 }
 
-export function destroyServer(location: string) {
-  if (servers[location]) {
-    for (const id of Object.keys(servers[location].streams)) {
-      destroyStream(id, location)
+export async function destroyServer(location: string) {
+  return new Promise<void>((resolve) => {
+    while (servers[location]) {
+      for (const id of Object.keys(servers[location].streams)) {
+        destroyStream(id, location)
+      }
+      delete servers[location]
```

Comment:
but by the time we resolve this all servers would be deleted (i think)

Reply:
You're welcome to experiment with other approaches.  All I can say is this seems to deterministically solve the issue of "server already exists at this location" errors intermittently appearing in the integration test.  It's just a brute force way of forcing the servers to get cleared.  I still think a long term better approach would be to somehow remove that global servers object that's in `./network.ts` or else revise our `destroy` method to just take an array of servers/services and then blow them all away and resolve once the servers object is empty